src-native/asami/main.clj

```diff
@@ -1,51 +1,105 @@
 (ns ^{:doc "Entry point for CLI"
       :author "Paula Gearon"}
   asami.main
-  (:require [clojure.string :as s]
-            [clojure.edn :as edn]
+  (:require [asami.core :as asami]
+            [asami.graph :as graph]
             [cheshire.core :as json]
+            [clojure.edn :as edn]
+            [clojure.java.io :as io]
             [clojure.pprint :refer [pprint]]
-            [asami.core :as asami]
-            [asami.graph :as graph])
+            [clojure.string :as s])
+  (:import (java.io PushbackReader))
   (:gen-class))
 
-(def reader-opts {:readers (assoc graph/node-reader 'a/r re-pattern)})
+(def eof
+  (reify))
+
+(defn eof? [x]
+  (identical? x eof))
+
+(def reader-opts
+  {:eof eof
+   :readers (assoc graph/node-reader 'a/r re-pattern)})
 
 (defn process-args
```

Comment:
Can leave for now, but feel free to use `clojure.tools.cli` if you like 🙂

Reply:
Sure thing. That'll be a separate PR. 👍